tests/dracula/test_run_topp_jnt_crt.py

```diff
@@ -11,36 +12,55 @@
     """Class for unit test."""
 
-    os.chdir(os.path.dirname(os.path.abspath(__file__)))
-    waypts_jnt = np.loadtxt("test_waypts_jnt_0.txt")
-    lim_ones = np.ones(waypts_jnt.shape[1])
+    paths = sorted(
+        glob.glob(
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)),
+                "test_waypts_crt_*.txt",
+            )
+        )
+    )
+    waypts_jnt_list = [
+        np.loadtxt(path.replace("_crt_", "_jnt_")) for path in paths
+    ]
+    waypts_crt_list = [np.loadtxt(path) for path in paths]
+    lim_ones = np.ones(waypts_jnt_list[0].shape[1])
     lim_ones = np.vstack([-lim_ones, lim_ones]).T
     vlim_jnt = 1 * lim_ones  # rad/s
     alim_jnt = 7.5 * lim_ones  # rad/s^2
-    waypts_crt = np.loadtxt("test_waypts_crt_0.txt")
-    lim_ones = np.ones(waypts_crt.shape[1])
+    lim_ones = np.ones(waypts_crt_list[0].shape[1])
     lim_ones = np.vstack([-lim_ones, lim_ones]).T
     vlim_crt = 1 * lim_ones  # m/s
     alim_crt = 10 * lim_ones  # m/s^2
 
     def test_run_topp_jnt_crt(self):
-        """Test run_topp_jnt_crt(), just assert no throw."""
-        cs = dracula.run_topp_jnt_crt(  # effectively assertNoRaise
-            self.waypts_jnt,
-            self.vlim_jnt,
-            self.alim_jnt,
-            self.waypts_crt,
-            self.vlim_crt,
-            self.alim_crt,
-            return_cs=True,
-        )
-        v_abs = np.abs(cs.derivative(1)(cs.x))
-        a_abs = np.abs(cs.derivative(2)(cs.x))
-        self.assertAlmostEqual(v_abs[0].max(), 0)
-        self.assertAlmostEqual(v_abs[-1].max(), 0)
-        self.assertAlmostEqual(a_abs[0].max(), 0)
-        self.assertAlmostEqual(a_abs[-1].max(), 0)
-        self.assertGreater(cs.x[-1], 0.479)
+        """Test case for run_topp_jnt_crt().
+
+        Test that run_topp_jnt_crt() runs through fine, that the first and
+        second derivatives are all zero at both ends, and that the new runtime
+        is indeed extended compared to the joint-space-only calculation for
+        the known test case due the the addition of Cartesian v & a limits.
+        """
+        for i, (waypts_jnt, waypts_crt) in enumerate(
+            zip(self.waypts_jnt_list, self.waypts_crt_list)
+        ):
+            print(f"Testing data file {i}...")
+            cs = dracula.run_topp_jnt_crt(  # effectively assertNoRaise
+                waypts_jnt,
+                self.vlim_jnt,
+                self.alim_jnt,
+                waypts_crt,
+                self.vlim_crt,
+                self.alim_crt,
+                return_cs=True,
+            )
+            v_abs = np.abs(cs.derivative(1)(cs.x))
+            a_abs = np.abs(cs.derivative(2)(cs.x))
+            self.assertAlmostEqual(v_abs[0].max(), 0)
+            self.assertAlmostEqual(v_abs[-1].max(), 0)
+            self.assertAlmostEqual(a_abs[0].max(), 0)
+            self.assertAlmostEqual(a_abs[-1].max(), 0)
+            self.assertGreater(cs.x[-1], 0.479)
```

Comment:
Shouldn't this be greater than or equal?

Reply:
no because for this particular known test case the duration is greater than that after taking into account cartesian limits but smaller without them